src/blocks/amd_gpu.rs

```diff
@@ -101,29 +111,69 @@ pub async fn run(config: &Config, api: &CommonApi) -> Result<()> {
     }
 }
 
+pub struct Device {
+    path: PathBuf,
+}
+
 struct GpuInfo {
     utilization_percents: f64,
     vram_total_bytes: f64,
     vram_used_bytes: f64,
 }
 
-async fn read_prop<T: FromStr>(device: &str, prop: &str) -> Option<T> {
-    read_file(format!("/sys/class/drm/{device}/device/{prop}"))
-        .await
-        .ok()
-        .and_then(|x| x.parse().ok())
-}
+impl Device {
+    fn new(name: &str) -> Self {
+        Self {
+            path: PathBuf::from(format!("/sys/class/drm/{name}/device")),
+        }
+    }
 
-async fn read_gpu_info(device: &str) -> Result<GpuInfo> {
-    Ok(GpuInfo {
-        utilization_percents: read_prop::<f64>(device, "gpu_busy_percent")
-            .await
-            .error("Failed to read gpu_busy_percent")?,
-        vram_total_bytes: read_prop::<f64>(device, "mem_info_vram_total")
-            .await
-            .error("Failed to read mem_info_vram_total")?,
-        vram_used_bytes: read_prop::<f64>(device, "mem_info_vram_used")
+    async fn default_card() -> std::io::Result<Option<Self>> {
+        let mut dir = read_dir("/sys/class/drm").await?;
+
+        while let Some(entry) = dir.next_entry().await? {
+            let name = entry.file_name();
+            let Some(name) = name.to_str() else { continue };
+            if !name.starts_with("card") {
+                continue;
+            }
+
+            let mut path = entry.path();
+            path.push("device");
+
+            let Ok(uevent) = read_file(path.join("uevent")).await else {
+                continue;
+            };
+
+            if uevent.contains("DRIVER=amdgpu") {
```

Comment:
Maybe niche but if using proprietary drivers for whatever reason then this might be different?

Another option could be to use the PCI_ID, 1002 should always be AMD: https://admin.pci-ids.ucw.cz/read/PC/1002

Reply:
Do other drivers provide the same sysfs interface?

> Another option could be to use the PCI_ID, 1002 should always be AMD

Sounds good.